src/mumble/pluginAPI/PluginComponents.h

```diff
@@ -0,0 +1,134 @@
+/**
+ * This header file contains definitions of types and other components used in Mumble's plugin system
+ */
+
+#ifndef MUMBLE_PLUGINCOMPONENT_H_
+#define MUMBLE_PLUGINCOMPONENT_H_
+
+#include <stdint.h>
+#include <stddef.h>
+
+#define STATUS_OK 0
+
+enum TalkingState {
+	INVALID=-1,
+	PASSIVE=0,
+	TALKING,
+	WHISPERING,
+	SHOUTING
+};
+
+struct Version {
+	int32_t major;
+	int32_t minor;
+	int32_t patch;
+};
+
+
+typedef enum TalkingState TalkingState_t;
```

Comment:
I don't think we need typedefs for classes/structs that we are already adding.
If we were to modify them, we'd just modify their content anyway.

Reply:
True :thinking:

Reply:
Ah wait - this typedef was for not having to use the `enum` prefix every time we refer to the enum type (as has to be done to be C-compatible).